asciidoc/session-API.adoc

```diff
@@ -0,0 +1,682 @@
+
+[.tabbed-example]
+.Auto-commit transactions
+========
+[.include-with-dotnet]
+======
+
+.Auto-commit transactions
+[source, csharp]
+----
+include::{dotnet-examples}/ExamplesAsync.cs[tags=async-autocommit-transaction]
+----
+======
+
+[.include-with-java]
+======
+
+.Auto-commit transactions
+[source, java]
+----
+include::{java-examples}/AsyncAutocommitTransactionExample.java[tags=async-autocommit-transaction]
+----
+======
+
+
+[.include-with-javascript]
+======
+
+.Auto-commit transactions
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=async-autocommit-transaction]
+----
+======
+========
+
+[[driver-async-result-consume]]
+=== Consuming results
+
+The asynchronous session API provides language-idiomatic methods to aid integration with asynchronous applications and frameworks.
+
+[.tabbed-example]
+.Consuming results
+========
+[.include-with-dotnet]
+======
+
+.Consuming results
+[source, csharp]
+----
+include::{dotnet-examples}/ExamplesAsync.cs[tags=async-result-consume]
+----
+======
+
+[.include-with-java]
+======
+
+.Consuming results
+[source, java]
+----
+include::{java-examples}/AsyncResultConsumeExample.java[tags=async-result-consume]
+----
+======
+
+
+[.include-with-javascript]
+======
+
+.Consuming results
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=async-result-consume]
+----
+======
+========
+
+[[driver-rx-sessions]]
+== Reactive Sessions
+
+[NOTE]
+Available in .NET, Java and JavaScript
+
+Starting with Neo4j 4.0, the reactive processing of queries is supported.
+This can be achieved through reactive sessions.
+Reactive sessions allow for dynamic management of the data that is being exchanged between the driver and the server.
+
+Typical of reactive programming, consumers control the rate at which they consume records from queries and the driver in turn manages the rate at which records are requested from the server.
+Flow control is supported throughout the entire Neo4j stack, meaning that the query engine responds correctly to the flow control signals.
+This results in far more efficient resource handling and ensures that the receiving side is not forced to buffer arbitrary amounts of data.
+
+For more information about reactive stream, please see the following:
+
+*  https://www.reactivemanifesto.org/[The Reactive Manifesto]
+*  https://www.reactive-streams.org[Reactive Streams for JVM and JavaScript]
+*  https://projectreactor.io/docs/core/release/reference/[Project Reactor reference documentation]
+*  http://introtorx.com/[Introduction to Reactive Extensions to .NET]
+
+
+=== Dependencies
+
+Reactive sessions will typically be used in a client application that is already oriented towards the reactive style; it is expected that a reactive dependency or framework is in place.
+
+Refer to <<driver-get-started>> for more information on our recommended dependencies.
```

Comment:
I propose to strike "our" in this sentence.

Reply:
Already gone.